sympy/solvers/solvers.py

```diff
@@ -398,23 +430,19 @@ def check_assumptions(expr, against=None, **assumptions):
        >>> z = Symbol('z')
        >>> check_assumptions(z, real=True)
     """
-    if against is not None:
-        assumptions = against.assumptions0
-
     expr = sympify(expr)
-
-    result = True
-    for key, expected in assumptions.items():
-        if expected is None:
-            continue
+    if against:
+        if not isinstance(against, Symbol):
+            raise TypeError('against should be of type Symbol')
+        if assumptions:
+            raise AssertionError('No assumptions should be specified')
+        assumptions = against.assumptions0
+    a = assumptions
+    flag = []
+    for key in a:
         test = getattr(expr, 'is_' + key, None)
-        if test is expected:
-            continue
-        elif test is not None:
-            return False
-        result = None  # Can't conclude, unless an other test fails.
-    return result
-
+        flag.append(None) if test is None else flag.append(test is a[key])
+    return True if all(flag) else False if False in flag else None
```

Comment:
I would suggest that you keep the quick-exit capabilities of the original. There is a fuzzy function that will handle this...is it `fuzzy_set`?

Reply:
> is it fuzzy_set?

It is [_fuzzy_group](https://github.com/sympy/sympy/blob/master/sympy/core/logic.py#L40). But I don't think it would work rather `fuzzy_and` will be better to use.
@smichr will it be okay to use?